src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -280,5 +280,108 @@ public static void ThrowNotSupportedException_DeserializeCreateObjectDelegateIsN
                 throw new NotSupportedException(SR.Format(SR.DeserializeMissingParameterlessConstructor, invalidType));
             }
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonException_MetadataValuesInvalidToken()
+        {
+            throw new JsonException(SR.MetadataInvalidTokenAfterValues);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonException_MetadataReferenceNotFound(string id)
+        {
+            throw new JsonException(SR.Format(SR.MetadataReferenceNotFound, id));
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonException_MetadataValueWasNotString()
+        {
+            throw new JsonException(SR.MetadataValueWasNotString);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonException_MetadataReferenceObjectCannotContainOtherProperties()
+        {
+            throw new JsonException(SR.MetadataReferenceCannotContainOtherProperties);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void ThrowJsonException_MetadataReferenceObjectCannotContainOtherProperties_Dictionary(ref ReadStackFrame current)
+        {
+            current.KeyName = null;
```

Comment:
Why is doing this necessary?

Reply:
So the JSON Path can point to the Parent object and not to the dictionary key.